CONTRIBUTING.rst

```diff
@@ -0,0 +1,214 @@
+enhancements submitted by others. The issues vary significantly in complexity
+and difficulty, so you can try to find something that matches your skill
+level and knowledge.
+
+Pull requests
+-------------
+
+On GitHub pull requests are the main mechanism to contribute code. They
+are easy to use both for the contributor and for person accepting the
+contribution, and with more complex contributions it is easy also for
+others to join the discussion. Preconditions for creating a pull
+requests are having a `GitHub account`_, installing `Git`_ and forking the
+`SeleniumLibrary project`_.
+
+GitHub has good articles explaining how to `set up Git`_, `fork a repository`_
+and `use pull requests`_ and we do not go through them in more detail.
+We do, however, recommend to create dedicated branches for pull requests
+instead of creating them based on the master branch. This is especially
+important if you plan to work on multiple pull requests at the same time.
+
+Coding conventions
+------------------
+
+SeleniumLibrary uses the general Python code conventions defined in
+`PEP-8`_. In addition to that, we try to write `idiomatic Python`_
+and follow the `SOLID principles`_. with all new code. An important guideline
+is that the code should be clear enough that comments are generally not needed.
+
+Docstrings should be added to public keywords but are not generally
+needed in internal code. When docstrings are added, they should follow
+`PEP-257`_. See `Documentation`_ section below for more details about
+documentation syntax, generating docs, etc.
+
+We are pretty picky about using whitespace. We use blank lines and
+whitespace in expressions as dictated by
+`PEP-8`_, but we also follow these rules:
+
+-  Indentation using spaces, not tabs.
+-  No trailing spaces.
+-  No extra empty lines at the end of the file.
+-  Files must end with a newline.
+
+The above rules are good with most other code too. Any good editor or
+IDE can be configured to automatically format files according to them.
+
+Documentation
+-------------
+
+With new features or enhancements adequate documentation is as important
+as the actual functionality. Different documentation is needed depending
+on the issue.
+
+The main source of documentation should be placed in to the library and
+individual keywords.
+
+-  Other keywords and sections in the library introduction can be
+   referenced with internal links created with backticks like \`Example
+   Keyword\`
+
+-  When referring to arguments, argument names must use in inline code
+   style created with double backticks like \`\`argument\`\`.
+
+-  Examples are recommended whenever the new keyword or enhanced
+   functionality is not trivial.
+
+-  All new enhancements or changes should have a note telling when the
+   change was introduced. Often adding something like
+   ``New in Selenium2Libray 1.8.`` is enough.
+
+Keyword documentation can be easily created using `invoke`_ task::
+
+    inv generate-documentation
```

Comment:
I prefer `invoke generate-documentation`

Reply:
Changed to `keyword_documentation`